apps/api/src/domain/models/song/commands/publish-song.command.integration.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { Ack, CommandHandlerService, FluxStandardAction } from '@coscrad/commands';
+import buildMockUuidGenerator from '../../../../app/controllers/command/__tests__/buildMockUuidGenerator';
+import setUpIntegrationTest from '../../../../app/controllers/__tests__/setUpIntegrationTest';
+import { InternalError } from '../../../../lib/errors/InternalError';
+import { ArangoConnectionProvider } from '../../../../persistence/database/arango-connection.provider';
+import generateRandomTestDatabaseName from '../../../../persistence/repositories/__tests__/generateRandomTestDatabaseName';
+import TestRepositoryProvider from '../../../../persistence/repositories/__tests__/TestRepositoryProvider';
+import { DTO } from '../../../../types/DTO';
+import getValidResourceInstanceForTest from '../../../domainModelValidators/__tests__/domainModelValidators/utilities/getValidResourceInstanceForTest';
+import { ResourceType } from '../../../types/ResourceType';
+import buildInMemorySnapshot from '../../../utilities/buildInMemorySnapshot';
+import InvalidCommandPayloadTypeError from '../../shared/common-command-errors/InvalidCommandPayloadTypeError';
+import buildDummyUuid from '../../__tests__/utilities/buildDummyUuid';
+import { PublishSong } from './publish-song.command';
+import { PublishSongCommandHandler } from './publish-song.command-handler';
+
+const publishSongCommandType = 'PUBLISH_SONG';
+
+const dummyUuid = buildDummyUuid();
+
+const validCommandFSA: FluxStandardAction<DTO<PublishSong>> = {
+    type: publishSongCommandType,
+    payload: {
+        id: dummyUuid,
+    },
+};
+
+const validPayload = validCommandFSA.payload;
+describe('PublishSong', () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let arangoConnectionProvider: ArangoConnectionProvider;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, arangoConnectionProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateRandomTestDatabaseName(),
+            }));
+
+        commandHandlerService.registerHandler(
+            publishSongCommandType,
+            new PublishSongCommandHandler(testRepositoryProvider, buildMockUuidGenerator())
+        );
+    });
+
+    afterAll(async () => {
+        await arangoConnectionProvider.dropDatabaseIfExists();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('when the payload is valid', () => {
+        it('should succeed', async () => {
+            await testRepositoryProvider.addFullSnapshot(
```

Comment:
One nice way to ease the burden of writing these tests would be to have some helpers:
- assertCommandSuccess(
initialState: InMemorySnapshot,
commandFSA: CommandFSA,
valiateFinalState: async () => void
)

- assertCommandFailure(
   initialState,
   checkErrorResult: (result: InternalError): void,
   checkFinalState: async () => void
 )

Reply:
Ok- let's do this on the next command integration test we write.